src/aspire/utils/preprocess.py

```diff
@@ -148,23 +148,38 @@ def downsample(insamples, szout, mask=None):
         # stack of one dimension objects
 
         for idata in range(ndata):
-            insamples_fft = crop_pad(fftshift(fft(insamples[idata])), L_out) * mask
-            outsamples[idata] = np.real(ifft(ifftshift(insamples_fft)) * (L_out / L_in))
+            imsample_shifted = fft.fftshift(fft.fft(xp.asarray(insamples[idata])))
```

Comment:
`insamples_shifted`

Reply:
Done in commit 308ea16.

Reply:
Now it says `imsamples_shifted`. It should be `insamples_shifted`.

Reply:
Sorry. Changed it and merged it with the previous commit.